## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 444054 --skip-package-regex python312Packages.*`
Commit: `6ec038bf592cbc3c9e1dac5c4bb054f2348aeb6f`

---
### `x86_64-linux`
<details>
  <summary>:x: 14 packages failed to build:</summary>
  <ul>
    <li>expliot</li>
    <li>expliot.dist</li>
    <li>python313Packages.control</li>
    <li>python313Packages.control.dist</li>
    <li>python313Packages.dipy</li>
    <li>python313Packages.dipy.dist</li>
    <li>python313Packages.firedrake</li>
    <li>python313Packages.firedrake.dist</li>
    <li>python313Packages.qpsolvers</li>
    <li>python313Packages.qpsolvers.dist</li>
    <li>python313Packages.scikit-survival</li>
    <li>python313Packages.scikit-survival.dist</li>
    <li>sofa</li>
    <li>xfitter</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 331 packages built:</summary>
  <ul>
    <li>aligator</li>
    <li>aligator.doc</li>
    <li>ants</li>
    <li>apbs</li>
    <li>appcsxcad</li>
    <li>bonmin</li>
    <li>c3d</li>
    <li>casadi</li>
    <li>ceres-solver</li>
    <li>ceres-solver.dev</li>
    <li>cholmod-extra</li>
    <li>cloudcompare</li>
    <li>crocoddyl</li>
    <li>crocoddyl.doc</li>
    <li>csxcad</li>
    <li>dartsim</li>
    <li>dolfinx</li>
    <li>elastix</li>
    <li>elmerfem</li>
    <li>exhibit</li>
    <li>ezminc</li>
    <li>f3d</li>
    <li>f3d.man</li>
    <li>freecad</li>
    <li>g2o</li>
    <li>g2o.debug</li>
    <li>g2o.dev</li>
    <li>gama</li>
    <li>gdcm</li>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>getdp</li>
    <li>gimp (gimpPlugins.gimp)</li>
    <li>gimp-with-plugins</li>
    <li>gimp.dev (gimpPlugins.gimp.dev)</li>
    <li>gimp3 (gimp3Plugins.gimp)</li>
    <li>gimp3-with-plugins</li>
    <li>gimp3.dev (gimp3Plugins.gimp.dev)</li>
    <li>gimp3.devdoc (gimp3Plugins.gimp.devdoc)</li>
    <li>gimp3Plugins.gmic</li>
    <li>gimp3Plugins.lightning</li>
    <li>gimpPlugins.bimp</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gimplensfun</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.resynthesizer</li>
    <li>gimpPlugins.texturize</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>gklib</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>haskellPackages.casadi-bindings-internal</li>
    <li>haskellPackages.casadi-bindings-internal.doc</li>
    <li>ieda</li>
    <li>intensity-normalization (python313Packages.intensity-normalization)</li>
    <li>intensity-normalization.dist (python313Packages.intensity-normalization.dist)</li>
    <li>ipopt</li>
    <li>itk (itk_5)</li>
    <li>itk_5_2</li>
    <li>jupyter-all</li>
    <li>kahip</li>
    <li>metis</li>
    <li>mim-solvers</li>
    <li>minc_widgets</li>
    <li>mirtk</li>
    <li>mrtrix</li>
    <li>mumps</li>
    <li>mumps-mpi</li>
    <li>naev</li>
    <li>ndcurves</li>
    <li>ndcurves.doc</li>
    <li>netgen (python313Packages.netgen-mesher)</li>
    <li>octave</li>
    <li>octaveFull</li>
    <li>octavePackages.arduino</li>
    <li>octavePackages.audio</li>
    <li>octavePackages.bim</li>
    <li>octavePackages.bsltl</li>
    <li>octavePackages.cgi</li>
    <li>octavePackages.communications</li>
    <li>octavePackages.control</li>
    <li>octavePackages.database</li>
    <li>octavePackages.dataframe</li>
    <li>octavePackages.dicom</li>
    <li>octavePackages.divand</li>
    <li>octavePackages.doctest</li>
    <li>octavePackages.financial</li>
    <li>octavePackages.fpl</li>
    <li>octavePackages.fuzzy-logic-toolkit</li>
    <li>octavePackages.ga</li>
    <li>octavePackages.general</li>
    <li>octavePackages.generate_html</li>
    <li>octavePackages.geometry</li>
    <li>octavePackages.gsl</li>
    <li>octavePackages.image</li>
    <li>octavePackages.image-acquisition</li>
    <li>octavePackages.instrument-control</li>
    <li>octavePackages.interval</li>
    <li>octavePackages.io</li>
    <li>octavePackages.linear-algebra</li>
    <li>octavePackages.lssa</li>
    <li>octavePackages.mapping</li>
    <li>octavePackages.matgeom</li>
    <li>octavePackages.miscellaneous</li>
    <li>octavePackages.msh</li>
    <li>octavePackages.mvn</li>
    <li>octavePackages.nan</li>
    <li>octavePackages.ncarray</li>
    <li>octavePackages.netcdf</li>
    <li>octavePackages.nurbs</li>
    <li>octavePackages.octclip</li>
    <li>octavePackages.octproj</li>
    <li>octavePackages.optics</li>
    <li>octavePackages.optiminterp</li>
    <li>octavePackages.quaternion</li>
    <li>octavePackages.queueing</li>
    <li>octavePackages.signal</li>
    <li>octavePackages.sockets</li>
    <li>octavePackages.splines</li>
    <li>octavePackages.statistics</li>
    <li>octavePackages.stk</li>
    <li>octavePackages.strings</li>
    <li>octavePackages.struct</li>
    <li>octavePackages.symbolic</li>
    <li>octavePackages.tsa</li>
    <li>octavePackages.video</li>
    <li>octavePackages.windows</li>
    <li>octavePackages.zeromq</li>
    <li>octopus</li>
    <li>openems</li>
    <li>openmvg</li>
    <li>openmvs</li>
    <li>openturns</li>
    <li>otb</li>
    <li>p4est</li>
    <li>p4est-dbg</li>
    <li>pagmo2</li>
    <li>paraview</li>
    <li>parmmg</li>
    <li>pcl</li>
    <li>petsc</li>
    <li>photonvision</li>
    <li>pinocchio</li>
    <li>pinocchio.doc</li>
    <li>precice</li>
    <li>proxsuite-nlp</li>
    <li>proxsuite-nlp.doc</li>
    <li>python313Packages.aerosandbox</li>
    <li>python313Packages.aerosandbox.dist</li>
    <li>python313Packages.aligator</li>
    <li>python313Packages.aligator.doc</li>
    <li>python313Packages.blackjax</li>
    <li>python313Packages.blackjax.dist</li>
    <li>python313Packages.casadi</li>
    <li>python313Packages.crocoddyl</li>
    <li>python313Packages.crocoddyl.doc</li>
    <li>python313Packages.cvxopt</li>
    <li>python313Packages.cvxopt.dist</li>
    <li>python313Packages.cvxpy</li>
    <li>python313Packages.cvxpy.dist</li>
    <li>python313Packages.cyipopt</li>
    <li>python313Packages.cyipopt.dist</li>
    <li>python313Packages.dartsim</li>
    <li>python313Packages.dcmstack</li>
    <li>python313Packages.dcmstack.dist</li>
    <li>python313Packages.deid</li>
    <li>python313Packages.deid.dist</li>
    <li>python313Packages.dicom-numpy</li>
    <li>python313Packages.dicom-numpy.dist</li>
    <li>python313Packages.dicom2nifti</li>
    <li>python313Packages.dicom2nifti.dist</li>
    <li>python313Packages.dicomweb-client</li>
    <li>python313Packages.dicomweb-client.dist</li>
    <li>python313Packages.example-robot-data</li>
    <li>python313Packages.example-robot-data.doc</li>
    <li>python313Packages.f3d</li>
    <li>python313Packages.f3d.man</li>
    <li>python313Packages.fast-simplification</li>
    <li>python313Packages.fast-simplification.dist</li>
    <li>python313Packages.fenics-dolfinx</li>
    <li>python313Packages.fenics-dolfinx.dist</li>
    <li>python313Packages.fslpy</li>
    <li>python313Packages.fslpy.dist</li>
    <li>python313Packages.gdcm</li>
    <li>python313Packages.gstools</li>
    <li>python313Packages.gstools.dist</li>
    <li>python313Packages.heudiconv</li>
    <li>python313Packages.heudiconv.dist</li>
    <li>python313Packages.highdicom</li>
    <li>python313Packages.highdicom.dist</li>
    <li>python313Packages.itk</li>
    <li>python313Packages.jaxopt</li>
    <li>python313Packages.jaxopt.dist</li>
    <li>python313Packages.kahip</li>
    <li>python313Packages.medpy</li>
    <li>python313Packages.medpy.dist</li>
    <li>python313Packages.medvol</li>
    <li>python313Packages.medvol.dist</li>
    <li>python313Packages.mim-solvers</li>
    <li>python313Packages.nanoeigenpy</li>
    <li>python313Packages.nanoeigenpy.dev</li>
    <li>python313Packages.nanoeigenpy.doc</li>
    <li>python313Packages.napari-nifti</li>
    <li>python313Packages.napari-nifti.dist</li>
    <li>python313Packages.ndcurves</li>
    <li>python313Packages.ndcurves.doc</li>
    <li>python313Packages.neuralfoil</li>
    <li>python313Packages.neuralfoil.dist</li>
    <li>python313Packages.nibabel</li>
    <li>python313Packages.nibabel.dist</li>
    <li>python313Packages.nilearn</li>
    <li>python313Packages.nilearn.dist</li>
    <li>python313Packages.nipy</li>
    <li>python313Packages.nipy.dist</li>
    <li>python313Packages.nipype</li>
    <li>python313Packages.nipype.dist</li>
    <li>python313Packages.nitime</li>
    <li>python313Packages.nitime.dist</li>
    <li>python313Packages.nitransforms</li>
    <li>python313Packages.nitransforms.dist</li>
    <li>python313Packages.niworkflows</li>
    <li>python313Packages.niworkflows.dist</li>
    <li>python313Packages.opensfm</li>
    <li>python313Packages.opensfm.dist</li>
    <li>python313Packages.openturns</li>
    <li>python313Packages.osqp</li>
    <li>python313Packages.osqp.dist</li>
    <li>python313Packages.pepit</li>
    <li>python313Packages.pepit.dist</li>
    <li>python313Packages.petsc4py</li>
    <li>python313Packages.picos</li>
    <li>python313Packages.picos.dist</li>
    <li>python313Packages.pinocchio</li>
    <li>python313Packages.pinocchio.doc</li>
    <li>python313Packages.proxsuite-nlp</li>
    <li>python313Packages.proxsuite-nlp.doc</li>
    <li>python313Packages.pybids</li>
    <li>python313Packages.pybids.dist</li>
    <li>python313Packages.pydicom</li>
    <li>python313Packages.pydicom.dist</li>
    <li>python313Packages.pygmo</li>
    <li>python313Packages.pykrige</li>
    <li>python313Packages.pykrige.dist</li>
    <li>python313Packages.pylibjpeg</li>
    <li>python313Packages.pylibjpeg-openjpeg</li>
    <li>python313Packages.pylibjpeg-openjpeg.dist</li>
    <li>python313Packages.pylibjpeg-rle</li>
    <li>python313Packages.pylibjpeg-rle.dist</li>
    <li>python313Packages.pylibjpeg.dist</li>
    <li>python313Packages.pymedio</li>
    <li>python313Packages.pymedio.dist</li>
    <li>python313Packages.pynetdicom</li>
    <li>python313Packages.pynetdicom.dist</li>
    <li>python313Packages.pyorthanc</li>
    <li>python313Packages.pyorthanc.dist</li>
    <li>python313Packages.pyotb</li>
    <li>python313Packages.pyotb.dist</li>
    <li>python313Packages.pyprecice</li>
    <li>python313Packages.pyprecice.dist</li>
    <li>python313Packages.pyradiomics</li>
    <li>python313Packages.pyradiomics.dist</li>
    <li>python313Packages.python-csxcad</li>
    <li>python313Packages.python-csxcad.dist</li>
    <li>python313Packages.pyvista</li>
    <li>python313Packages.pyvista.dist</li>
    <li>python313Packages.qutip</li>
    <li>python313Packages.qutip.dist</li>
    <li>python313Packages.scikits-odes</li>
    <li>python313Packages.scikits-odes-sundials</li>
    <li>python313Packages.scikits-odes-sundials.dist</li>
    <li>python313Packages.scikits-odes.dist</li>
    <li>python313Packages.simpleitk</li>
    <li>python313Packages.simpleitk.dist</li>
    <li>python313Packages.slepc4py</li>
    <li>python313Packages.sleqp</li>
    <li>python313Packages.templateflow</li>
    <li>python313Packages.templateflow.dist</li>
    <li>python313Packages.torchWithRocm</li>
    <li>python313Packages.torchWithRocm.cxxdev</li>
    <li>python313Packages.torchWithRocm.dev</li>
    <li>python313Packages.torchWithRocm.dist</li>
    <li>python313Packages.torchWithRocm.lib</li>
    <li>python313Packages.torchio</li>
    <li>python313Packages.torchio.dist</li>
    <li>python313Packages.trx-python</li>
    <li>python313Packages.trx-python.dist</li>
    <li>python313Packages.tsid</li>
    <li>python313Packages.tsid.doc</li>
    <li>python313Packages.vtk</li>
    <li>python313Packages.xnatpy</li>
    <li>python313Packages.xnatpy.dist</li>
    <li>qcsxcad</li>
    <li>qcsxcad.dev</li>
    <li>rocmPackages.hipsolver (rocmPackages_6.hipsolver)</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip, rocmPackages_6.mivisionx, rocmPackages_6.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu (rocmPackages_6.mivisionx-cpu)</li>
    <li>rtabmap</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>scipopt-gcg</li>
    <li>scipopt-scip</li>
    <li>scipopt-ug</li>
    <li>simpleitk</li>
    <li>slepc</li>
    <li>sleqp</li>
    <li>spral</li>
    <li>suitesparse (suitesparse_5_3)</li>
    <li>suitesparse.dev (suitesparse_5_3.dev)</li>
    <li>suitesparse.doc (suitesparse_5_3.doc)</li>
    <li>sundials</li>
    <li>sundials.examples</li>
    <li>therion</li>
    <li>toppler</li>
    <li>trilinos</li>
    <li>trilinos-mpi</li>
    <li>tsid</li>
    <li>tsid.doc</li>
    <li>vpv</li>
    <li>vtk (vtk_9_5)</li>
    <li>vtk-dicom</li>
    <li>vtk-full</li>
    <li>vtkWithQt5</li>
    <li>vtkWithQt6</li>
    <li>xyce</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>xyce.doc</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python313Packages.control</summary>
<pre>    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
control/tests/docstrings_test.py:23: in &lt;module&gt;
    import numpydoc.docscrape as npd
E   ModuleNotFoundError: No module named &#x27;numpydoc&#x27;
___________________ ERROR collecting doc/test_sphinxdocs.py ____________________
ImportError while importing test module &#x27;/build/source/doc/test_sphinxdocs.py&#x27;.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/829wb290i87wngxlh404klwxql5v18p4-python3-3.13.7/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
doc/test_sphinxdocs.py:15: in &lt;module&gt;
    import numpydoc.docscrape as npd
E   ModuleNotFoundError: No module named &#x27;numpydoc&#x27;
=========================== short test summary info ============================
ERROR control/tests/docstrings_test.py
ERROR doc/test_sphinxdocs.py
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 2 errors in 3.08s ===============================</pre>
</details>
<details>
<summary>python313Packages.dipy</summary>
<pre>FAILED: [code=1] dipy/denoise/enhancement_kernel.cpython-313-x86_64-linux-gnu.so.p/enhancement_kernel.c 
/nix/store/jigf1i6ixbrnijnbq7x1ikvhjggw0v7m-python3.13-cython-3.1.2/bin/cython -3 --fast-fail --warning-errors --output-file dipy/denoise/enhancement_kernel.cpython-313-x86_64-linux-gnu.so.p/enhancement_kernel.c --include-dir /build/source/.mesonpy-pysjaxem/dipy ../dipy/denoise/enhancement_kernel.pyx

Error compiling Cython file:
------------------------------------------------------------
...
        rng = np.random.default_rng()

        if isinstance(orientations, Sphere):
            # use the sphere defined by the user
            sphere = orientations
        elif isinstance(orientations, (int, long, float)):
                                            ^
------------------------------------------------------------

/build/source/dipy/denoise/enhancement_kernel.pyx:81:44: undeclared name not builtin: long
[40/123] Generating &#x27;dipy/align/vector_fields.cpython-313-x86_64-linux-gnu.so.p/vector_fields.c&#x27;iterion.c&#x27;c&#x27;ction_getter.c&#x27;[K
ninja: build stopped: subcommand failed.

ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
<details>
<summary>python313Packages.scikit-survival</summary>
<pre>dependency /nix/store/xq23bm8sim4q9jbmb1dbfmwg9hrjwpkj-python3.13-numpy-2.3.2/lib/python3.13/site-packages/numpy/_core/include/numpy/arrayobject.h won&#x27;t be automatically included in the manifest: the path must be relative
dependency /nix/store/xq23bm8sim4q9jbmb1dbfmwg9hrjwpkj-python3.13-numpy-2.3.2/lib/python3.13/site-packages/numpy/_core/include/numpy/arrayscalars.h won&#x27;t be automatically included in the manifest: the path must be relative
dependency /nix/store/xq23bm8sim4q9jbmb1dbfmwg9hrjwpkj-python3.13-numpy-2.3.2/lib/python3.13/site-packages/numpy/_core/include/numpy/ndarrayobject.h won&#x27;t be automatically included in the manifest: the path must be relative
dependency /nix/store/xq23bm8sim4q9jbmb1dbfmwg9hrjwpkj-python3.13-numpy-2.3.2/lib/python3.13/site-packages/numpy/_core/include/numpy/ndarraytypes.h won&#x27;t be automatically included in the manifest: the path must be relative
dependency /nix/store/xq23bm8sim4q9jbmb1dbfmwg9hrjwpkj-python3.13-numpy-2.3.2/lib/python3.13/site-packages/numpy/_core/include/numpy/ufuncobject.h won&#x27;t be automatically included in the manifest: the path must be relative
reading manifest file &#x27;scikit_survival.egg-info/SOURCES.txt&#x27;
reading manifest template &#x27;MANIFEST.in&#x27;
no previously-included directories found matching &#x27;doc/_build&#x27;
no previously-included directories found matching &#x27;doc/api/generated&#x27;
warning: no previously-included files matching &#x27;__pycache__&#x27; found anywhere in distribution
warning: no previously-included files matching &#x27;.ipynb_checkpoints&#x27; found anywhere in distribution
warning: no previously-included files matching &#x27;*.py[oc]&#x27; found anywhere in distribution
warning: no previously-included files matching &#x27;*.bak&#x27; found anywhere in distribution
warning: no previously-included files matching &#x27;*.swp&#x27; found anywhere in distribution
warning: no previously-included files matching &#x27;*~&#x27; found anywhere in distribution
adding license file &#x27;COPYING&#x27;
writing manifest file &#x27;scikit_survival.egg-info/SOURCES.txt&#x27;

ERROR Missing dependencies:
        scikit-learn~=1.6.1; python_version &lt;= &quot;3.13&quot;</pre>
</details>
<details>
<summary>sofa</summary>
<pre>This warning is for project developers.  Use -Wno-dev to suppress it.

-- Adding plugin MultiThreading
-- Adding plugin SofaMatrix
CMake Warning at Sofa/framework/Config/cmake/Modules/Findmetis.cmake:54 (message):
  metis version 5.2.1 found in
  /nix/store/06ah132ckg3jadqg84rqav02qsiwpxvy-metis-5.2.1/include, but exact
  version 5.1.0 is required
Call Stack (most recent call first):
  Sofa/framework/Config/cmake/Modules/Findmetis.cmake:83 (_metis_check_version)
  applications/plugins/SofaMatrix/CMakeLists.txt:11 (find_package)


CMake Error at applications/plugins/SofaMatrix/CMakeLists.txt:21 (message):
  SofaMatrix: DEPENDENCY metis NOT FOUND.  SOFA_ALLOW_FETCH_DEPENDENCIES is
  OFF and thus cannot be fetched.  Install metis (version=5.1.0), or enable
  SOFA_ALLOW_FETCH_DEPENDENCIES to fix this issue.


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

--- a/pkgs/by-name/gk/gklib/package.nix
+++ b/pkgs/by-name/gk/gklib/package.nix
@@ -8,13 +8,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "gklib";
-  version = "5.1.1-unstable-2023-03-27";
+  version = "5.1.1-unstable-2025-07-15";
 
   src = fetchFromGitHub {
     owner = "KarypisLab";
     repo = "GKlib";
-    rev = "8bd6bad750b2b0d90800c632cf18e8ee93ad72d7";
-    hash = "sha256-tunepMLaRDR5FQVL/9S7/w6e1j+f2+pg01H/0/z/ZCI=";
+    rev = "6e7951358fd896e2abed7887196b6871aac9f2f8";
+    hash = "sha256-jT0hT5Y3E8GnE8OJWzDj5rtz9s59sMEXLduUnBV0I0Y=";
   };
 
   nativeBuildInputs = [ cmake ];

--- a/pkgs/by-name/me/metis/package.nix
+++ b/pkgs/by-name/me/metis/package.nix
@@ -1,34 +1,49 @@
 {
   lib,
   stdenv,
-  fetchurl,
-  unzip,
+  fetchFromGitHub,
+  fetchpatch,
   cmake,
+  gklib,
+  llvmPackages,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "metis";
-  version = "5.1.0";
+  version = "5.2.1";
 
-  src = fetchurl {
-    url = "http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-${version}.tar.gz";
-    sha256 = "1cjxgh41r8k6j029yxs8msp3z6lcnpm16g5pvckk35kc7zhfpykn";
+  src = fetchFromGitHub {
+    owner = "KarypisLab";
+    repo = "METIS";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-eddLR6DvZ+2LeR0DkknN6zzRvnW+hLN2qeI+ETUPcac=";
   };
 
-  cmakeFlags = [
-    "-DGKLIB_PATH=../GKlib"
-    # remove once updated past https://github.com/KarypisLab/METIS/commit/521a2c360dc21ace5c4feb6dc0b7992433e3cb0f
-    "-DCMAKE_SKIP_BUILD_RPATH=ON"
+  patches = [
+    # fix gklib link error
+    (fetchpatch {
+      url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/sci-libs/metis/files/metis-5.2.1-add-gklib-as-required.patch?id=c78ecbd3fdf9b33e307023baf0de12c4448dd283";
+      hash = "sha256-uoXMi6pMs5VrzUmjsLlQYFLob1A8NAt9CbFi8qhQXVQ=";
+    })
   ];
-  nativeBuildInputs = [
-    unzip
-    cmake
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ gklib ] ++ lib.optional stdenv.cc.isClang llvmPackages.openmp;
+
+  preConfigure = ''
+    make config
+  '';
+
+  cmakeFlags = [
+    (lib.cmakeBool "OPENMP" true)
+    (lib.cmakeBool "SHARED" (!stdenv.hostPlatform.isStatic))
   ];
 
   meta = {
     description = "Serial graph partitioning and fill-reducing matrix ordering";
-    homepage = "http://glaros.dtc.umn.edu/gkhome/metis/metis/overview";
+    homepage = "https://github.com/KarypisLab/METIS";
     license = lib.licenses.asl20;
     platforms = lib.platforms.all;
   };
-}
+})

--- a/pkgs/by-name/pa/parmetis/package.nix
+++ b/pkgs/by-name/pa/parmetis/package.nix
@@ -3,31 +3,35 @@
   stdenv,
   fetchFromGitHub,
   cmake,
+  gklib,
   metis,
   mpi,
 }:
 
 stdenv.mkDerivation {
   pname = "parmetis";
-  version = "4.0.3";
+  version = "4.0.3-unstable-2023-03-26";
 
   src = fetchFromGitHub {
     owner = "KarypisLab";
     repo = "ParMETIS";
-    rev = "d90a2a6cf08d1d35422e060daa28718376213659";
-    hash = "sha256-22YQxwC0phdMLX660wokRgmAif/9tRbUmQWwNMZ//7M=";
+    rev = "8ee6a372ca703836f593e3c450ca903f04be14df";
+    hash = "sha256-L9SLyr7XuBUniMH3JtaBrUHIGzVTF5pr014xovQf2cI=";
   };
 
   nativeBuildInputs = [ cmake ];
   enableParallelBuilding = true;
-  buildInputs = [ mpi ];
+  buildInputs = [
+    gklib
+    metis
+    mpi
+  ];
 
   configurePhase = ''
     runHook preConfigure
 
-    tar xf ${metis.src}
-    mv metis-* metis
-    make config metis_path=metis gklib_path=metis/GKlib prefix=$out
+    make config metis_path=${metis} gklib_path=${gklib} prefix=$out \
+       shared=${if stdenv.hostPlatform.isStatic then "0" else "1"}
 
     runHook postConfigure
   '';

